src/igor/sync.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (2018) Sandia Corporation.
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+
+package main
+
+import (
+	"fmt"
+	log "minilog"
+)
+
+var cmdSync = &Command{
+	UsageLine: "sync",
+	Short:     "synchronize igor data",
+	Long: `
+Does an internal check to verify the integrity of the data file.
+
+OPTIONAL FLAGS:
+
+	-v	Verbose - Prints additioanl information
+	`,
+}
+
+var subL bool // -l
+var subV bool // -v
+
+func init() {
+	// break init cycle
+	cmdSync.Run = runSync
+
+	cmdSync.Flag.BoolVar(&subL, "l", false, "")
+	cmdSync.Flag.BoolVar(&subV, "v", false, "")
+}
+
+// Gather data integrity information, report, and fix
+func runSync(cmd *Command, args []string) {
+	user, err := getUser()
+	if err != nil {
+		log.Fatalln("Cannot determine current user", err)
+	}
+	if user.Username != "root" {
+		log.Fatalln("Sync access restricted. Please use as admin.")
+	}
+
+	log.Debug("Sync called - finding orphan IDs")
+	IDs := getOrphanIDs()
+	if len(IDs) > 0 {
+		fmt.Printf("%v orphan Reservation IDs found\n", len(IDs))
+		if subV {
+			for _, id := range IDs {
+				fmt.Println(id)
+			}
+		}
+	}
+	// we are only listing
+	if subL {
+		return
+	}
+
+	// purge the orphan IDs from the shedule
+	if len(IDs) > 0 {
+		if !subV {
+			fmt.Println("Purging Orphan IDs from Schedule...")
+		}
+		for _, oid := range IDs {
+			purgeFromSchedule(oid)
+		}
+		fmt.Println("Done.")
+		dirty = true
+	}
+
+}
+
+func getOrphanIDs() []uint64 {
+	resIDs := make(map[uint64]bool)
+	// make a list of all reseration IDs that appear in the schedule
+	for _, s := range Schedule {
+		for _, n := range s.Nodes {
+			resIDs[n] = true
+		}
+	}
+	// Go through the reservations and turn off IDs we know about
+	for _, r := range Reservations {
+		resIDs[r.ID] = false
```

Comment:
Instead you could `delete(resIDs, r.ID)` so that below you don't need to check the value of resIDs for the given `k` -- it should only be false.